components/block/element.js

```diff
@@ -9,62 +10,112 @@ export default class Block extends LitElement {
+
+    this.model.on('sidebar_save:success', this.refresh.bind(this));
   }
 
-  edit(e){
-    console.log(e.target.innerText)
-    this.refresh();
+  get layout() {
+    return window.parent.Core.g.layout;
   }
 
-  delete(e) {
-    console.log(e.target.innerText)
-    this.remove();
+  get model() {
+    return (this.cached_model ||= this.layout.blocks.findWhere({
+      id: this.blockId,
+    }));
   }
 
-  moveUp(e) {
-    console.log(e.target.innerText)
-    const prevElement = this.previousElementSibling;
-    isBlock(prevElement) && this.parentNode.insertBefore(this, prevElement)
+  get parent() {
+    if (this.cached_parent !== undefined) return this.cached_parent;
+
+    const parentId = this.model.attributes.parent_block_id;
+    const parent = document.querySelector(`ngl-block[blockId="${parentId}"]`);
+
+    if (parent) return (this.cached_parent = parent);
+    else return (this.cached_parent = null);
   }
 
-  moveDown(e) {
-    console.log(e.target.innerText)
-    const nextElement = this.nextElementSibling;
-    isBlock(nextElement) && this.parentNode.insertBefore(nextElement, this)
+  get isInLinkedZone() {
+    return this.model.zone().is_linked();
+  }
+
+  renderMenu() {
+    if (this.isInLinkedZone) return this.renderLinkedBlockMenu();
+    else if (this.parent) return this.renderInnerBlockMenu();
+    else return this.renderOuterBlockMenu();
+  }
+
+  renderLinkedBlockMenu() {
+    return html` <button @click=${this.refresh}>Refresh</button> `;
+  }
+
+  renderInnerBlockMenu() {
+    return html`
+      ${this.renderOuterBlockMenu()}
+      <button @click=${this.parentEdit}>Edit container</button>
+      <button @click=${this.parentRefresh}>Refresh container</button>
+    `;
+  }
+
+  renderOuterBlockMenu() {
+    return html`
+      <button @click=${this.edit}>Edit</button>
+      <button @click=${this.refresh}>Refresh</button>
+    `;
+  }
+
+  edit() {
+    this.model.trigger('edit');
+  }
+
+  parentEdit() {
+    this.parent.edit();
   }
 
   async fetch() {
-    const resp = await fetch(window.location.href)
-    return resp.text()
+    this.loading = true;
+    const resp = await fetch(window.location.href);
+    this.loading = false;
+    return resp.text();
   }
 
   async refresh() {
     let html = await this.fetch();
-    const template = document.createElement('template')
+
+    const template = document.createElement('template');
     template.innerHTML = html;
-    const currentBlockHtml = template.content.querySelector(`ngl-block[blockId="${this.blockId}"]`)
+    const currentBlockHtml = template.content.querySelector(
+      `ngl-block[blockId="${this.blockId}"]`
+    );
 
-    // Simulate text change
-    currentBlockHtml.querySelector('.timestamp').innerText = new Date().toLocaleTimeString();
     this.innerHTML = currentBlockHtml.innerHTML;
+
+    this.dispatchEvent(
+      new Event('ngl:refresh', {bubbles: true, composed: true})
+    );
+  }
+
+  async parentRefresh() {
+    await this.parent.refresh();
```

Comment:
Remove async, await as it's not needed here.

await is only needed if you do something with promise response, and async is only needed if you use await inside the function.

Reply:
Removed in f4602e9bd9225f12b6fb88d62b34ba95b82b62eb.
The promise from `parent.refresh()` can can easily be returned from `parentRefresh` if it ever becomes necessary to `await` it at some point.

Reply:
Exactly!